nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/migrator/AbstractRepositoryMigrator.java

```diff
@@ -84,7 +117,18 @@ protected void migrateRepository(String repositoryName) {
 
     protected abstract String probeSession(CoreSession session);
 
-    protected abstract void migrateSession(CoreSession session);
+    /**
+     * @deprecated since 11.1 Use {@link #migrateSession(String, MigrationContext, CoreSession)} instead
+     */
+    @Deprecated
+    protected void migrateSession(CoreSession session) {
+        migrateSession(null, null, session);
+    }
+
+    /**
+     * @since 11.1
+     */
+    protected abstract void migrateSession(String step, MigrationContext context, CoreSession session);
```

Comment:
This can't be backported as it because people having their own migrator won't be able to install the hotfix without drawbacks. Furthermore, its migrator should keep on working.

With these constraints, for 10.10, you should remove the abstract declaration and make this method calling the previous one (no need to deprecate stuff for 10.10 if it looks weird). The previous could throw an exception as base implementation, as it you will be able to not implements them in our migrators. Same for `migrateRepository`.
You'll need a new `step` field to do that.

Reply:
We reworked the classes with the step field to avoid drawbacks.
Please check out again